### PR TITLE
GEODE-6450: ClientClusterManagementService supports ssl connection

### PR DESCRIPTION
**Pull request by jinmeiliao**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by jinmeiliao** [comment]
@Petahhh

File: geode-management/src/main/java/org/apache/geode/management/internal/ClientClusterManagementService.java [hunk 1]
```diff
@@ -54,34 +73,91 @@ private ClientClusterManagementService() {
...
+  }
+
+  public ClientClusterManagementService(ClientHttpRequestFactory requestFactory) {
+    this();
+    this.restTemplate.setRequestFactory(requestFactory);
+  }
+
+  public static SSLContext createTrustAllSSLContext() {
+    try {
+      SSLContext sslContext = SSLContext.getInstance("TLS");
```
Comment (jdeppe-pivotal) [inline geode-management/src/main/java/org/apache/geode/management/internal/ClientClusterManagementService.java]: This should probably be using `SSLContext.getDefault()`. That way it gets initialized automatically and will also honor the various `javax.net.ssl.*` properties if set. Please make sure there is a pure client test that sets and uses these properties.

File: geode-core/src/main/java/org/apache/geode/management/internal/api/GeodeClusterManagementServiceFactory.java [hunk 2]
```diff
@@ -74,45 +77,38 @@ public ClusterManagementService create() throws IllegalStateException {
       Set<InternalDistributedMember> locatorsWithClusterConfig =
           cacheImpl.getDistributionManager().getAllHostedLocatorsWithSharedConfiguration()
               .keySet();
-      String serviceAddress = getHttpServiceAddress(locatorsWithClusterConfig);
 
-      return new ClientClusterManagementService(serviceAddress);
+      MemberInformation memberInformation = getHttpServiceAddress(locatorsWithClusterConfig);
+
+      return create(getHostName(memberInformation), memberInformation.getHttpServicePort(),
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/api/GeodeClusterManagementServiceFactory.java]: At this point you need to ensure you're picking up and using the SSL options that have been set on this server.

**Comment by jinmeiliao** [comment]
failed test is a flaky test not related to this changeset

**Pull request merged**